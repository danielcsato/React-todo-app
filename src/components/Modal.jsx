import React from 'react';

const Modal = ({ setIsOpen, deleteTodo }) => {
  return (
    <>
      <div className="overlay" onClick={() => setIsOpen(false)} />
      <div className="modal">
        <div className="modalTitle">
          <h5>Are you sure?</h5>
        </div>
        <div className="modalButtons">
          <button className="modalKeep" onClick={() => setIsOpen(false)}>
            Keep
          </button>
          <button className="modalDelete" onClick={deleteTodo}>
            Delete
          </button>
        </div>
      </div>
    </>
  );
};

export default Modal;
