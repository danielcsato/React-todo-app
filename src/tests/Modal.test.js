import React from 'react';
import renderer from 'react-test-renderer';

import Modal from '../components/Modal';

it('renders correctly', () => {
  const component = renderer.create(<Modal />).toJSON();
  expect(component).toMatchSnapshot();
});
