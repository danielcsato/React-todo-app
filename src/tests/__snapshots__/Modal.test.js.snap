// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="overlay"
    onClick={[Function]}
  />,
  <div
    className="modal"
  >
    <div
      className="modalTitle"
    >
      <h5>
        Are you sure?
      </h5>
    </div>
    <div
      className="modalButtons"
    >
      <button
        className="modalKeep"
        onClick={[Function]}
      >
        Keep
      </button>
      <button
        className="modalDelete"
        data-test-id="confirmTrashBtn"
      >
        Delete
      </button>
    </div>
  </div>,
]
`;
